### PR TITLE
refactor: `PolarGrid` to function component

### PR DESCRIPTION
**Pull request by akamfoad**
### rm: JSDoc eslint rule in favor of typescript

JSDoc is great for documenting code in JS, but since the library built with typescript natively, we shall use JSDoc only for documenting purposes. For everything related to typing is better to utilize TypeScript to keep the library source code consistent and correct. The ESLint rule for valid JSDoc comments have been removed since it requires all comments to have some properties that would be the duplicate of the TS types and adds maintenance overhead with no value. Also this rule [has been deprecated 5y ago](https://eslint.org/blog/2018/11/jsdoc-end-of-life/)

### refactor: `PolarGrid` to function component

Converted the PolarGrid component to functional component and converted the static functions to in-component functions, then refactored them out of the component itself. After that converted all outside functions which rendered JSX to be React components instead.

## Motivation and Context

Refactoring library to be more maintainable - testing refactor-ability

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

**Comment by ckifer** [comment]
Looks like the build id failing

**Review by ckifer — CHANGES_REQUESTED** [review]
The rest looks good to me, but build is failing and lets keep the ESLint rule for now. Agreed with you, but we can modify in a different PR

**Comment by ckifer** [comment]
I am hesitant to merge this one due to it existing in this map https://github.com/recharts/recharts/blob/e7858c926b01e05f846168eed093821a89aa2c68/src/chart/generateCategoricalChart.tsx#L2140-L2161

This means that the static property `displayName` is used by the `generateCategoricalChart` function. That means more potential for bugs (imo) in the components in that map as compared to the others. Just a callout

**Comment by akamfoad** [comment]
@ckifer I understand, tbh I tried to refactor some other components but I hesitated too since it or some of it's static members were used in this function to generate the charts. I think we need to refactor that somehow, I might try again once more, will keep this map in mind for the next refactor.

**Comment by ckifer** [comment]
Totally agreed, the more I understand how it's working the more I consider it an anti-pattern. 

Recharts in general needs an architectural deep dive and maybe some flow diagrams so we can figure out a decently replacement for the static member issue among others.

**Comment by nikolasrieble** [comment]
Please separate this into two PRs, one for the refactoring and one for the eslint rule. Both are separate concerns. Removing the eslint rule still seems a reasonable improvement.

**Comment by akamfoad** [comment]
@nikolasrieble I removed the ESLint change at the time, this PR only changes PolarGrid

File: src/polar/PolarGrid.tsx [hunk 1]
```diff
@@ -18,159 +18,146 @@ interface PolarGridProps {
...
-      <g className="recharts-polar-grid-concentric">
-        {polarRadius.map((entry: number, i: number) =>
-          gridType === 'circle' ? this.renderConcentricCircle(entry, i) : this.renderConcentricPolygon(entry, i),
-        )}
-      </g>
-    );
+  return (
+    <g className="recharts-polar-grid-concentric">
+      {polarRadius.map((entry: number, i: number) => {
+        const key = i;
```
Comment (ckifer) [inline src/polar/PolarGrid.tsx:139]: nit: why assign a variable to `i` which is already defined?

**Review by ckifer — APPROVED** [review]
Overall looks good. Can we do the same kind of comparison as with `Curve` before merge?

File: src/polar/PolarGrid.tsx (hunk 1 above)
Comment (akamfoad) [reply src/polar/PolarGrid.tsx:139]: Actually just to trick ESLint regarding the `no-array-index-key` rule, that otherwise shows a warning.

**Comment by akamfoad** [comment]
Of course @ckifer here it is:

```
recharts on refactor/polar-grid-component ➜ pixelmatch .\master-polar-grid.png .\refactored-polar-grid.png diff.png 0.1

matched in: 14.547ms
different pixels: 0
error: 0%
```

### On `master`
master-polar-grid



### On `akamfoad:refactor/polar-grid-component`
refactored-polar-grid


### `Diff`
diff

**Comment by ckifer** [comment]
Wonderful, thank you!

**Pull request merged**